### PR TITLE
multi download part 1

### PR DESCRIPTION
**Pull request by pgrudra**

### CODE REVIEW

**Review by yash10019coder — CHANGES_REQUESTED** [review]
Hi @pgrudra left some comments PTAL thanks.

File: app/src/main/java/org/zus/helloworld/ui/vult/VultFragment.kt [hunk 1]
```diff
@@ -880,7 +901,7 @@ class VultFragment : Fragment(), FileClickListener, ThumbnailDownloadCallback {
             CoroutineScope(vultViewModel.viewModelScope.coroutineContext).launch {
                 vultViewModel.listFiles("/")
             }
-            showSuccessSnackbar();
+            showSuccessSnackbar()
```
[the remark this answers is not in the document]
Comment (pgrudra) [reply app/src/main/java/org/zus/helloworld/ui/vult/VultFragment.kt:904]: The success message is about the file being uploaded successfully. Even if listFiles fails for some reason, the file is uploaded successfully when completed() is called, so it's okay to have it this way

File: app/build.gradle [hunk 2]
```diff
@@ -26,6 +26,7 @@ android {
     }
     buildFeatures {
         viewBinding true
+        dataBinding true
```
[the remark this answers is not in the document]
Comment (pgrudra) [reply app/build.gradle:29]: Ops, was trying out something, but forgot to remove this. Will remove soon

File: app/src/main/res/layout/vult_fragment.xml [hunk 3]
```diff
@@ -1,7 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.swiperefreshlayout.widget.SwipeRefreshLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
+    xmlns:tools="http://schemas.android.com/tools">
+    <data>
+        <variable
+            name="vultViewModel"
+            type="org.zus.helloworld.ui.vult.VultViewModel" />
```
[the remark this answers is not in the document]
Comment (pgrudra) [reply app/src/main/res/layout/vult_fragment.xml:8]: Yep, removing these in next commit

File: app/src/main/java/org/zus/helloworld/ui/vult/VultFragment.kt [hunk 4]
```diff
@@ -607,7 +618,7 @@ class VultFragment : Fragment(), FileClickListener, ThumbnailDownloadCallback {
     private fun previewAction(position: Int, next: Boolean) {
         var position = position
         if (position < 0) position =
-            filesAdapter!!.itemCount - 1 else if (position == filesAdapter!!.getItemCount()) position =
+            filesAdapter!!.itemCount - 1 else if (position == filesAdapter!!.itemCount) position =
```
[the remark this answers is not in the document]
Comment (pgrudra) [reply app/src/main/java/org/zus/helloworld/ui/vult/VultFragment.kt:621]: Not necessary

File: app/src/main/java/org/zus/helloworld/ui/vult/VultFragment.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (yash10019coder) [reply app/src/main/java/org/zus/helloworld/ui/vult/VultFragment.kt:934]: got it

**Review by yash10019coder — COMMENTED** [review]
LGTM @pgrudra merging it.

**Pull request merged**